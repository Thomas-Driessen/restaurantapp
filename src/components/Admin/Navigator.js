import React from 'react';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import FastFoodIcon from '@material-ui/icons/Fastfood';
import LocalCafeIcon from '@material-ui/icons/LocalCafe';
import Input from '@material-ui/core/Input';
import Button from '@material-ui/core/Button';
import MenuBookIcon from '@material-ui/icons/MenuBook';
import CategoryIcon from '@material-ui/icons/Category';
import KitchenIcon from '@material-ui/icons/Kitchen';
import BarChartIcon from '@material-ui/icons/BarChart';

class Navigator extends React.Component {
    constructor() {
        super();
        this.state = {
            restaurantNameData: {title: 'Restaurant'},
            restaurantNameButtonDisable: true,
            newName: null
        }
    }
    componentDidMount() {
        let mounted = true;
        if (mounted) {
            this.getRestaurantName()
        }
        return () => mounted = false;
    }

    getRestaurantName = () => {
        fetch(`${process.env.REACT_APP_API_URL}/api/RestaurantName`, {
            method: 'get',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
        }).then(response => response.json())
            .then(data => {
                console.log(data)
                this.setState({restaurantNameData: data})
            });
    }

    renderSaveButton = (changeEvent) =>{
        if(this.state.restaurantNameButtonDisable === true){
            this.setState({restaurantNameButtonDisable: false});
        }
        this.setState({newName: changeEvent.target.value});
    }

    saveRestaurantName = () =>{
        var newRestaurantName = {
            "id": this.state.restaurantNameData.id,
            "title": this.state.newName
        };
        fetch(`${process.env.REACT_APP_API_URL}/api/RestaurantName`, {
            method: 'put',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(newRestaurantName)
        }).then(response => response.json())
            .then(data => {
                console.log(data)
            });
        this.setState({restaurantNameButtonDisable: true});
    }

    render() {
        return (
            <Drawer variant="permanent">
                <List>
                    <ListItem>
                        <Input 
                            placeholder={this.state.restaurantNameData.title}
                            type='text'
                            onChange={this.renderSaveButton}
                        />
                        <Button onClick={this.saveRestaurantName} disabled={this.state.restaurantNameButtonDisable}>
                            Save
                        </Button>
                    </ListItem>
                    <ListItem>

                    </ListItem>
                    <ListItem>
                        <Button onClick={this.props.selectFoods}>
                            <FastFoodIcon /> Foods
                        </Button>
                    </ListItem>
                    <ListItem>
                        <Button onClick={this.props.selectDrinks}>
                            <LocalCafeIcon /> Drinks
                        </Button>
                    </ListItem>
                    <ListItem>
                        <Button onClick={this.props.selectFoodCategories}>
                            <CategoryIcon /> Food Categories
                        </Button>
                    </ListItem>
                    <ListItem>
                        <Button onClick={this.props.selectDrinkCategories}>
                            <CategoryIcon /> Drink Categories
                        </Button>
                    </ListItem>
                    <ListItem>
                        <Button onClick={this.props.selectIngredients}>
                            <KitchenIcon /> Ingredients
                        </Button>
                    </ListItem>
                    <ListItem>
                        <Button onClick={this.props.selectNotOnMenu}>
                            <MenuBookIcon /> Not On Menu
                        </Button>
                    </ListItem>
                    <ListItem>
                        <Button onClick={this.props.selectShowCharts}>
                            <BarChartIcon /> Charts
                        </Button>
                    </ListItem>
                </List>
            </Drawer>
        )
    }
}

export default Navigator;
