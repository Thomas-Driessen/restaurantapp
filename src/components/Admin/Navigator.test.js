import React from 'react';
import { render } from '@testing-library/react';
import Navigator from './Navigator';
import renderer from 'react-test-renderer';

test('Render Navigator Restaurant', () => {
    const { getByText } = render(<Navigator />);
    const linkElement = getByText(/Restaurant/i);
    expect(linkElement).toBeInTheDocument();
});

test('Render Navigator Dashboard', () => {
    const { getByText } = render(<Navigator />);
    const linkElement = getByText(/Dashboard/i);
    expect(linkElement).toBeInTheDocument();
});

it('Navigator renders correctly', () => {
    const tree = renderer
        .create(<Navigator />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

afterAll(() => {
    global.gc && global.gc()
})