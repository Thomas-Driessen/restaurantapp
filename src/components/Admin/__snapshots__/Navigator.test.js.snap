// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator renders correctly 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked"
>
  <div
    className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
  >
    <ul
      className="MuiList-root MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        Restaurant
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
             Dashboard
          </span>
        </button>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      />
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18.06 22.99h1.66c.84 0 1.53-.64 1.63-1.46L23 5.05h-5V1h-1.97v4.05h-4.97l.3 2.34c1.71.47 3.31 1.32 4.27 2.26 1.44 1.42 2.43 2.89 2.43 5.29v8.05zM1 21.99V21h15.03v.99c0 .55-.45 1-1.01 1H2.01c-.56 0-1.01-.45-1.01-1zm15.03-7c0-8-15.03-8-15.03 0h15.03zM1.02 17h15v2h-15z"
              />
            </svg>
             Foods
          </span>
        </button>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 3H4v10c0 2.21 1.79 4 4 4h6c2.21 0 4-1.79 4-4v-3h2c1.11 0 2-.9 2-2V5c0-1.11-.89-2-2-2zm0 5h-2V5h2v3zM4 19h16v2H4z"
              />
            </svg>
             Drinks
          </span>
        </button>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2l-5.5 9h11z"
              />
              <circle
                cx="17.5"
                cy="17.5"
                r="4.5"
              />
              <path
                d="M3 13.5h8v8H3z"
              />
            </svg>
             Food Categories
          </span>
        </button>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2l-5.5 9h11z"
              />
              <circle
                cx="17.5"
                cy="17.5"
                r="4.5"
              />
              <path
                d="M3 13.5h8v8H3z"
              />
            </svg>
             Drink Categories
          </span>
        </button>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 2.01L6 2c-1.1 0-2 .89-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.11-.9-1.99-2-1.99zM18 20H6v-9.02h12V20zm0-11H6V4h12v5zM8 5h2v3H8zm0 7h2v5H8z"
              />
            </svg>
             Ingredients
          </span>
        </button>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 5c-1.11-.35-2.33-.5-3.5-.5-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5V6c-.6-.45-1.25-.75-2-1zm0 13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5v11.5z"
              />
              <path
                d="M17.5 10.5c.88 0 1.73.09 2.5.26V9.24c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99zM13 12.49v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26V11.9c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.3-4.5.83zM17.5 14.33c-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26v-1.52c-.79-.16-1.64-.24-2.5-.24z"
              />
            </svg>
             Not On Menu
          </span>
        </button>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
              />
            </svg>
             Theme
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;
